pkg/secrets-store/provider_client.go

```diff
@@ -154,6 +155,36 @@ func (p *PluginClientBuilder) Cleanup() {
 	p.conns = make(map[string]*grpc.ClientConn)
 }
 
+// EnableHealthCheck enables periodic healthcheck for configured provider clients by making
+// a Version() RPC call. If the provider healthcheck fails, we log an error.
+func (p *PluginClientBuilder) EnableHealthCheck(interval time.Duration, stopCh <-chan struct{}) {
```

Comment:
Since this actually performs the health check and blocks indefinitely, I think `HealthCheck` may be a better name.

It would also be helpful to describe the blocking nature in the function comment so that caller knows stopping conditions.